daemon/daemon.go

```diff
@@ -918,11 +920,19 @@ func (daemon *Daemon) Shutdown() error {
 	}
 
 	for platform, ds := range daemon.stores {
-		if ds.layerStore != nil {
-			if err := ds.layerStore.Cleanup(); err != nil {
-				logrus.Errorf("Error during layer Store.Cleanup(): %v %s", err, platform)
+		go func() {
+			for {
+				if daemon.mountingContainers == 0 {
+					break
+				}
+				<-time.After(10 * time.Millisecond)
```

Comment:
Is `sync.Cond` suitable for this condition? If we use `cond`, the logic may be more complex,
Each mount during starting container should lock and increase mountContainers. When done, will check `mountingCountainers==0` if true, call `cond.Signal`,

I think more wait in `Shutdown` should be better. it is not called that much frequently.